From 49d0e45c2a7dcac58a806afb85524893acc6d05b Mon Sep 17 00:00:00 2001
From: Zhao Liang <leo.zhao@intel.com>
Date: Fri, 23 Aug 2013 15:43:23 +0800
Subject: Enable sync mode recording for timelapse mode

BZ: 131021

Category: aosp improvement
Domain: Video.Media-video recording
Origin: internal
Upstream-Candidate: no

Change-Id: I11354cd02048efdc2ac8ad6acb2902715cb491a2
Orig-Change-Id: Iaa11ed18a925b9d9b442fd69c3ed5792d400b948
Signed-off-by: Zhao Liang <leo.zhao@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 media/libstagefright/OMXCodec.cpp |   22 ++++++++++++++++++++++
 1 files changed, 22 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index f4ec02a..49a17db 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -609,6 +609,28 @@ status_t OMXCodec::configureCodec(const sp<MetaData> &meta) {
 
     initOutputFormat(meta);
 
+    if ((mFlags & kOnlySubmitOneInputBufferAtOneTime)
+            && !strncmp(mComponentName, "OMX.Intel.VideoEncoder", 22)) {
+
+        OMX_INDEXTYPE index;
+
+        status_t err = mOMX->getExtensionIndex(mNode,
+                               "OMX.Intel.index.enableSyncEncoding", &index);
+
+        if (err != OK) {
+            return err;
+        }
+
+        OMX_BOOL enable = OMX_TRUE;
+        err = mOMX->setParameter(mNode, index, (void*)&enable, sizeof(enable));
+
+        if (err != OK) {
+            CODEC_LOGE("SetParameter OMX.Intel.index.enableSyncEncoding"
+                       "returned error 0x%08x", err);
+            return err;
+        }
+    }
+
     if ((mFlags & kClientNeedsFramebuffer)
             && !strncmp(mComponentName, "OMX.SEC.", 8)) {
         // This appears to no longer be needed???
-- 
1.7.4.1

